dan_layer/template_lib/src/models/component.rs

```diff
@@ -96,6 +96,7 @@ pub struct ComponentHeader {
     pub access_rules: AccessRules,
     // TODO: Split the state from the header
     pub state: ComponentBody,
+    // pub associated_public_key: Option<RistrettoPublicKeyBytes>,
```

Comment:
Will we need this, or can it be removed ?

Reply:
Not sure, still have to think about how to associate a public key used for shard DH key derivation with an account component. Will remove for now.